src/core/tests/unit/vcfbed/test_vcftools.py

```diff
@@ -0,0 +1,258 @@
+
+    # non_hmer_fn testing
+    non_hmer_fn = vcftools.FilterWrapper(data).get_non_h_mer().get_fn().get_df()
+    assert all(non_hmer_fn["indel"])
+    assert (non_hmer_fn["hmer_indel_length"] == 0).all()
+    assert all(
+        [  # noqa: C419
+            row["classify"] == "fn"
+            or (row["classify"] == "tp" and (row["filter"] == "LOW_SCORE") and (row["filter"] != "PASS"))
+            for index, row in non_hmer_fn.iterrows()
+        ]
+    )
+
+
+def test_bed_output_when_no_tree_score(
+    resources_dir,
+):  # testing the case when there is no tree_score and there is blacklist
+    data = pd.read_hdf(pjoin(resources_dir, "exome.h5"), key="concordance")
+    df = vcftools.FilterWrapper(data)  # noqa: PD901
+    result = dict(df.get_fn().bed_format(kind="fn").get_df()["itemRgb"].value_counts())
+    expected_result = {
+        vcftools.FilteringColors.BLACKLIST.value: 169,
+        vcftools.FilteringColors.CLEAR.value: 89,
+        vcftools.FilteringColors.BORDERLINE.value: 39,
+    }
+    for k in result:
+        assert result[k] == expected_result[k]
+
+    df = vcftools.FilterWrapper(data)  # noqa: PD901
+    # since there is no tree_score all false positives should be the same color
+    result = dict(df.get_fp().bed_format(kind="fp").get_df()["itemRgb"].value_counts())
+
+    assert len(result.keys()) == 1
+
+
+def test_get_region_around_variant():
+    vpos = 100
+    vlocs = []
+    assert vcftools.get_region_around_variant(vpos, vlocs, 10) == (95, 105)
+
+
+class TestGetVcfDf:
+    def test_get_vcf_df(self, resources_dir):
+        input_vcf = pjoin(resources_dir, "test_get_vcf_df.vcf.gz")
+        df = vcftools.get_vcf_df(input_vcf)  # noqa: PD901
+        non_nan_columns = list(df.dropna(axis=1, how="all").columns)
+        non_nan_columns.sort()
+        assert non_nan_columns == [
+            "ac",
+            "ad",
+            "af",
+            "alleles",
+            "an",
+            "baseqranksum",
+            "chrom",
+            "db",
+            "dp",
+            "excesshet",
+            "filter",
+            "filtered_haps",
+            "fs",
+            "gnomad_af",
+            "gq",
+            "gt",
+            "hapcomp",
+            "hapdom",
+            "hec",
+            "id",
+            "indel",
+            "mleac",
+            "mleaf",
+            "mq",
+            "mqranksum",
+            "pl",
+            "pos",
+            "qd",
+            "qual",
+            "readposranksum",
+            "ref",
+            "sb",
+            "sor",
+            "tree_score",
+            "variant_type",
+            "x_css",
+            "x_gcc",
+            "x_ic",
+            "x_il",
+            "x_lm",
+            "x_rm",
+        ]
+
+    def test_get_vcf_df_use_qual(self, resources_dir):
+        input_vcf = pjoin(resources_dir, "test_get_vcf_df.vcf.gz")
+        df = vcftools.get_vcf_df(input_vcf, scoring_field="QUAL")  # noqa: PD901
```

Comment:
why not to fix this [PD901](https://docs.astral.sh/ruff/rules/pandas-df-variable-name/) error? here and in other places